checkout-core/src/main/java/com/adyen/checkout/core/internal/util/RunCompileOnly.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2023 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by oscars on 8/5/2023.
+ */
+
+package com.adyen.checkout.core.internal.util
+
+import androidx.annotation.RestrictTo
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+inline fun <R> runCompileOnly(block: () -> R): R? {
+    try {
+        return block()
```

Comment:
One limitation of this is if the block itself can return `null` right?

Reply:
> The default upper bound (if there was none specified) is `Any?` ([source](https://kotlinlang.org/docs/generics.html#generic-constraints))

So by default `R` is nullable here and `block` is allowed to return `null`. If we specify `<R : Any>` instead `block` can no longer return null

Reply:
Maybe it's better to force it to be non nullable? Or make `runCompileOnly` return a sealed class instead of null or not null? I'm fine with either.